pymc_experimental/model_builder.py

```diff
@@ -0,0 +1,299 @@
+import pymc as pm
+import arviz
+import pandas as pd
+from pathlib import Path
+import numpy as np
+from typing import Dict
+import cloudpickle
+import arviz as az
+import hashlib
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+
+class ModelBuilder(pm.Model):
+        """
+    Extention of pm.Model class to improve workflow.
+
+    ModelBuilder class can be used to play around models with ease using direct API calls
+    for multiple tasks that one need to deploy a model.  
+    
+    """
+
+    _model_type = "BaseClass"
+    version = "None"
+
+    def __init__(self, model_config: Dict, sampler_config: Dict):
+        """
+        Initialises model configration and sampler configration for the model
+        Parameters
+        ----------
+        model_confid: Dictionary
+            dictonary of parameters that initialise model configration. Genrated by the user defiend create_sample_input method.
+        sampler_config: Dictionary
+            dictonary of parameters that initialise sampler configration. Genrated by the user defiend create_sample_input method.
+        
+        Example:
+        >>> class LinearModel(ModelBuilder)
+            ...
+        >>> model = LinearModel(model_config, sampler_config)
+        """
+
+        super().__init__()
+        self.model_config = model_config  # parameters for priors etc.
+        self.sample_config = sampler_config  # parameters for sampling
+        self.idata = None  # parameters for
+
+    def _data_setter(
+        self, data: Dict[str, Union[np.ndarray, pd.DataFrame, pd.Series]], x_only: bool = True
+    ):
+        """
+        Sets new data in the model.
+
+        Parameter
+        --------
+
+        data: Dictionary of string and either of numpy array, pandas dataframe or pandas Series
+            It is the data we need to set as idata for the model
+        x_only: bool
+            if data only contains values of x and y is not present in the data
+
+        Example: 
+
+        >>>def _data_setter(self, data : pd.DataFrame):
```

Comment:
Again, wrong indents, please be more mindful here.
```suggestion
        >>> def _data_setter(self, data : pd.DataFrame):
```